### PR TITLE
New GRPC server + PT routing knobs/defaults; datadog stats; performance tests 🚀 

### PR DESCRIPTION
**Pull request by mark-idleman**
Suite of knob-adding, knob-turning, and monitoring improvements to speed up performance + visibility. Born out of the great router fires of Spring '21.

- Uncomments + expands datadog stats sent from servers (to compliment Python stats we already had). Java-side stats are tagged with release name and region, just like the Python stats
- Sets important preprocessing knob, the max walking time considered when interpolating all possible transfers from GTFS, to a default setting of 300 seconds (5 mins)
- Explicitly sets several "new" transit routing knobs for Graphhopper that we just didn't know about/hadn't set ourselves in the past. These include a limit on the total walk time allowed for a request (speeds things up significantly), knobs that tune preferences around how much people prefer walking + how to weight # of transfers when considering response quality (decided on by humans messing around in the GUI), and changing the transit routing algorithm from a pareto query (optimizes for both a low # of transfers and the shortest total trip time) to a soonest-arrival-time algorithm (only tries to find shortest overall trips; is faster performance-wise; we can use the `betaTransfers` knob to adjust how many transfers are being returned fairly well)
- Adds these new knobs to the GUI (except for `usePareto` but checkboxes in React are HARD, OK?!)
- Adds a bunch of tweak-ability for GRPC server settings (defined in `RouterServer.java`), all of which are setable via env vars (this is how we set them from the python side). We also define sane defaults for all of these in case they aren't set.
- Lowers thread counts for a bunch of the side-machinery we have setup to get the GUI working
- Adds 2 new complete performance benchmark tests, one of which just asks for routes inside Java using Graphhopper-internal routing objects, and one that creates a Java GRPC client and hits a locally-running GRPC router service

### CODE REVIEW

**Comment by JacobHayes** [comment]
@marcopalmeri tagging you as the nearest non-Mark Java person 😁

**Review by kaelgreco — APPROVED** [review]
I'm happy to merge these now since we've been using them already. we can update as we need

Only request (now or make a ticket and handle it later): explain the performance tests (how to run, what they do) in code or a readme

File: default_gh_config.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 graphhopper:
   datareader.file:
   gtfs.file:
-  gtfs.max_transfer_interpolation_walk_time_seconds: 600
+  gtfs.max_transfer_interpolation_walk_time_seconds: 300
```
[the remark this answers is not in the document]
Comment (mark-idleman) [reply default_gh_config.yaml:7]: That's basically it! But it's not only a time dependency (ex, how long you have to wait at a single point for the next bus to pick you up) - it's the max walking time graphhopper allows when interpolating new transfers to include during preprocessing (not taking into account wait time). For GTFS feeds with a `transfers.txt` file, it keeps all the transfers included in that file, and then adds all other possible inter- and intra-feed transfers that are under the walking time limit set here. For feeds w/o a transfers file, it interpolates all inter- and intra-feed transfers with the same walking time cutoff.

There's another knob that's set during queries that limits the total amount of walking time allowed overall for routes returned by the PT router, but that's slightly separate - at that point, all the transfers have already been precomputed (based on `max_transfer_interpolation_walk_time_seconds`)

File: grpc/src/main/java/RouterImpl.java [hunk 2]
```diff
@@ -237,20 +264,25 @@ public void info(InfoRequest request, StreamObserver<InfoReply> responseObserver
...
+                request.getEarliestDepartureTime().getSeconds(), request.getEarliestDepartureTime().getNanos()
+        ));
         ghPtRequest.setLimitSolutions(request.getLimitSolutions());
         ghPtRequest.setLocale(Locale.US);
         ghPtRequest.setArriveBy(false);
         ghPtRequest.setPathDetails(Lists.newArrayList("stable_edge_ids"));
         ghPtRequest.setProfileQuery(true);
         ghPtRequest.setMaxProfileDuration(Duration.ofMinutes(request.getMaxProfileDuration()));
         ghPtRequest.setBetaWalkTime(request.getBetaWalkTime());
+        ghPtRequest.setLimitStreetTime(Duration.ofSeconds(request.getLimitStreetTimeSeconds()));
```
[the remark this answers is not in the document]
Comment (mark-idleman) [reply grpc/src/main/java/RouterImpl.java:283]: Good call! They aren't well documented in the official Graphhopper docs, really, and we definitely don't have them documented well anywhere, so getting these up in the readme seems like a good call. [Ticket](https://replicahq.atlassian.net/browse/RAD-2460)

File: grpc/src/main/java/RouterImpl.java [hunk 3]
```diff
@@ -237,20 +264,25 @@ public void info(InfoRequest request, StreamObserver<InfoReply> responseObserver
...
+        ghPtRequest.setIgnoreTransfers(!request.getUsePareto()); // ignoreTransfers=true means pareto queries are off
```
[the remark this answers is not in the document]
Comment (mark-idleman) [reply grpc/src/main/java/RouterImpl.java:284]: Ah, this is a bit confusing - it is actually the case that "the condition of ignoreTransfers being true causes pareto queries to be turned off" 🙃 It turns out that `ignoreTransfers` is just a really badly named variable inside the graphhopper codebase - a better name would be `dont_use_pareto` or something, to capture how it actually works. The reason for the name: when you turn on pareto queries, the algorithm takes into account the number of transfers as one of the criteria it uses to decide which paths to return; when it's off, you're "ignoring" transfers, ie not factoring them in and only returning results based on which route gets you there the earliest.

So, in our proto files, we want to use a much clearer boolean to express this, `use_pareto`, but inside graphhopper, the opposite switch needs to be flipped, basically

**Pull request merged**